src/schema/sources.ts

```diff
@@ -573,6 +575,13 @@ export const canAccessSource = async (
   validateRankAgainstId?: string,
 ): Promise<boolean> => {
   if (permission === SourcePermissions.View && !source.private) {
+    if (sourceTypesWithMembers.includes(source.type)) {
+      const isMemberBlocked = member?.role === SourceMemberRoles.Blocked;
+      const canAccess = !isMemberBlocked;
+
+      return canAccess;
+    }
```

Comment:
More detailed alternative to this would be to create a set of permissions for anonymous users and then use those when calling `ensureSourcePermission` on feed, source, post queries. 

All of those currently just check `View` permission on source so this makes sure that even though source is public blocked member is still checked.

Reply:
I think for the use case at hand we simply want to say if you are blocked from public you don't even have view rights, so would cover as is.

When we get more requests we can evaluate separate rules for members and anonymous maybe?

Reply:
What would be the tradeoffs between the two?

Reply:
We missed it in planning so  main trade-off is time we would need to properly define how it would work and propagate that through all the API mutations.

Reply:
So let's keep it in mind for next time :)
We are already a bit behind schedule here